docs/metrics/iou.md

```diff
@@ -0,0 +1,170 @@
+# Intersection over Union (IoU)
+
+Intersection over Union (IoU) measures the ratio of the intersection and the union between two instances, ranging from 0 to 1 where 1 indicates a perfect match.  It is one of the metrics that measure the similarity between two instances, often used to compare predictions to ground truths.
+
+As the name suggests, the IoU of two instances ($A$ and $B$) is defined as:
+
+$$IoU(A, B) = \frac {A \cap B} {A \cup B}$$
+
+
+## When Do I Use IoU?
+It is often used to compare two geometries (e.g., bounding boxes, polygons) or segmentation masks in object detection, instance segmentation, or semantic segmentation models. In multi-label classification, IoU, more likely known as the [Jaccard index](https://en.wikipedia.org/wiki/Jaccard_index), is used to compare set of prediction labels for a sample to the corresponding set of ground truth labels. Moreover, there are workflows such as [action detection](https://paperswithcode.com/task/action-detection) and [video moment retrieval](https://paperswithcode.com/task/moment-retrieval) where IoU measures the **temporal** overlap between two time-series snippets.
+
+
+Because IoU can be used on various types of data, let's look at how the metric is defined for some of these data types:
+
+- [**2D Axis-Aligned Bounding Boxes**](#2d-axis-aligned-bounding-box)
+- [**Segmentation Mask**](#segmentation-mask)
+- [**Set of Labels**](#set-of-labels)
+
+
+### 2D Axis-Aligned Bounding Box
+
+Let's consider two 2D axis-aligned bounding boxes, $\text{A}$ and $\text{B}$, and notice the highlighted overlap region, which is always going to be a 2D axis-aligned bounding box.
+
+![An example of a 2D axis-aligned bounding box](../assets/images/metrics-iou-2dbbox.png)
+
+
+In order to compute IoU for two 2D axis-aligned bounding boxes, the first step is identifying the area of the intersection box, $(\text{A} \cap \text{B})$. The two coordinates of the intersection box, top-left and bottom-right corners, can be defined as:
```

Comment:
Could probably omit the `2D axis-aligned` modifier here, as it's clear from context